dvc/lock.py

```diff
@@ -56,13 +56,10 @@ def _do_lock(self):
             raise LockError(FAILED_TO_LOCK_MESSAGE)
 
     def lock(self):
-        try:
-            self._do_lock()
-            return
-        except LockError:
-            time.sleep(DEFAULT_TIMEOUT)
-
-        self._do_lock()
+        retries = 6
+        delay = DEFAULT_TIMEOUT / retries
+        lock_retry = retry(retries, LockError, timeout=delay)(self._do_lock)
```

Comment:
`delay` is clearly incorrect terminology :slightly_smiling_face: Let's just
```suggestion
        lock_retry = retry(RETRIES, LockError, timeout=TIMEOUT)(self._do_lock)
```
with, e.g.
```
TIMEOUT = 0.1
RETRIES = 50
```

Reply:
Why do you think so? 
From [funcy's documentation](https://funcy.readthedocs.io/en/stable/flow.html#retry):
> There will be a delay in `timeout` seconds between tries

Reply:
@skshetry Precisely because it is called `timeout` and not `delay`. Dividing stuff is not obvious, better to keep explicit TIMEOUT and RETRIES (and not hardcode it).

Reply:
Looks like timeout has two different meanings? I was only familiar with one.

Reply:
@skshetry They are pretty much the same, but you are thinking about the total lock timeout, while I'm talking about this particular piece of code, where timeout is about the time between tries.

Reply:
@efiop, `HardlinkLock` uses `DEFAULT_TIMEOUT` as well. I don't want to have two different timeouts, so I'd have to do something like following:
```suggestion
        delay = DEFAULT_TIMEOUT / RETRIES
        lock_retry = retry(RETRIES, LockError, timeout=delay)(self._do_lock)
```

which is no different from this PR.